src/components/common/Pagination.tsx

```diff
@@ -0,0 +1,72 @@
+import { cn } from '@/utils/cn';
+
+import Icon from './Icon';
+
+interface PaginationProps {
+  currentPage: number; // 현재 보고있는 페이지
+  totalCount: number; // 총 게시글 수
+  pageSize: number; // 한 페이지에 보여줄 게시글 수
+  onPageChange: (page: number) => void; // 페이지 변경 시 실행될 함수
+}
+
+export default function Pagination({
+  currentPage,
+  totalCount,
+  pageSize,
+  onPageChange,
+}: PaginationProps) {
+  const totalPages = Math.ceil(totalCount / pageSize);
+  const currentGroup = Math.floor((currentPage - 1) / 5);
+  const firstPage = currentGroup * 5 + 1;
+  const lastPage = Math.min(firstPage + 4, totalPages);
+  const pageCount = Math.max(0, lastPage - firstPage + 1);
+
+  const pageNumbers = new Array(pageCount).fill(0).map((_, i) => firstPage + i);
```

Comment:
```suggestion
  const pageNumbers = new Array(Math.max(0, lastPage - firstPage + 1)).fill(0).map((_, i) => firstPage + i);
```
💬 한 번만 쓰이는 변수라면 이렇게 써도 될 것 같습니다! 그리고 최솟값이 1이 아닌가 싶은데 `Math.max(0, lastPage - firstPage + 1)`로 쓰신 이유가 있을까요?